internal/service/lexv2models/confirmation_setting_equality.go

```diff
@@ -0,0 +1,179 @@
+				return item.MapBlockKey.ValueString() == key
+			})
+
+			if index != -1 {
+				if !arePromptAttemptValuesEqual(*newPromptAttemptSpecification[index], *value) {
+					return false, diags
+				}
+			} else if _, ok := promptAttemptSpecificationDefaults(ctx, key); ok {
+				hasDefaults = true
+				continue
+			}
+		}
+
+		return hasDefaults, diags
+	}
+
+	return false, diags
+}
+
+// arePromptAttemptValuesEqual compares two PromptAttemptsSpecification items for equality
+// ignoring the map_block_key field since it's used as the map key
+func arePromptAttemptValuesEqual(oldItem, newItem PromptAttemptsSpecification) bool {
+	return oldItem.AllowInterrupt.Equal(newItem.AllowInterrupt) &&
+		oldItem.AllowedInputTypes.Equal(newItem.AllowedInputTypes) &&
+		oldItem.AudioAndDTMFInputSpecification.Equal(newItem.AudioAndDTMFInputSpecification) &&
+		oldItem.TextInputSpecification.Equal(newItem.TextInputSpecification)
+}
+
+// promptAttemptSpecificationDefaults returns the default PromptAttemptsSpecification for a given key
+func promptAttemptSpecificationDefaults(ctx context.Context, key string) (*PromptAttemptsSpecification, bool) {
+	objectDefault := func(mapBlockKey PromptAttemptsType) *PromptAttemptsSpecification {
+		return &PromptAttemptsSpecification{
+			MapBlockKey: fwtypes.StringEnumValue(mapBlockKey),
+			AllowedInputTypes: fwtypes.NewListNestedObjectValueOfSliceMust(ctx, []*AllowedInputTypes{
+				{
+					AllowAudioInput: fwflex.BoolValueToFramework(ctx, true),
+					AllowDTMFInput:  fwflex.BoolValueToFramework(ctx, true),
+				},
+			}),
+			AllowInterrupt: fwflex.BoolValueToFramework(ctx, true),
+			AudioAndDTMFInputSpecification: fwtypes.NewListNestedObjectValueOfSliceMust(ctx, []*AudioAndDTMFInputSpecification{
+				{
+					StartTimeoutMs: fwflex.Int64ValueToFramework(ctx, 4000), //nolint:mnd
+					AudioSpecification: fwtypes.NewListNestedObjectValueOfSliceMust(ctx, []*AudioSpecification{
+						{
+							EndTimeoutMs: fwflex.Int64ValueToFramework(ctx, 640),   //nolint:mnd
+							MaxLengthMs:  fwflex.Int64ValueToFramework(ctx, 15000), //nolint:mnd
+						},
+					}),
+					DTMFSpecification: fwtypes.NewListNestedObjectValueOfSliceMust(ctx, []*DTMFSpecification{
+						{
+							DeletionCharacter: fwflex.StringValueToFramework(ctx, "*"),
+							EndCharacter:      fwflex.StringValueToFramework(ctx, "#"),
+							EndTimeoutMs:      fwflex.Int64ValueToFramework(ctx, 5000), //nolint:mnd
+							MaxLength:         fwflex.Int64ValueToFramework(ctx, 513),  //nolint:mnd
+						},
+					}),
+				},
+			}),
+			TextInputSpecification: fwtypes.NewListNestedObjectValueOfSliceMust(ctx, []*TextInputSpecification{
+				{
+					StartTimeoutMs: fwflex.Int64ValueToFramework(ctx, 30000), //nolint:mnd
+				},
+			}),
+		}
+	}
+
+	defaults := map[string]*PromptAttemptsSpecification{
+		"Initial": objectDefault("Initial"),
+		"Retry1":  objectDefault("Retry1"),
+	}
```

Comment:
I vaguely recall the number of returned `RetryN` objects being proportional to the `max_retries` argument. This map may need to be adjusted to return a variable number of keys based on configured retries.

Edit: I was able to confirm this by creating an intent with `v6.0.0` and setting `max_retries` set to `2` with no `prompt_attempt_specifications`. The subsequent plan includes a removal for the `Retry2` map key.

```
              - prompt_attempts_specification {
                  - allow_interrupt = true -> null
                  - map_block_key   = "Retry2" -> null
```